src/Facets/StargateFacetV2.sol

```diff
@@ -0,0 +1,166 @@
+            payable(msg.sender),
+            _stargateData.fee.nativeFee
+        );
+
+        _startBridge(_bridgeData, _stargateData);
+    }
+
+    /// PRIVATE METHODS ///
+
+    /// @dev Contains the business logic for the bridging via StargateV2
+    /// @param _bridgeData Data used purely for tracking and analytics
+    /// @param _stargateData Data specific to Stargate Bridge
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        StargateData calldata _stargateData
+    ) private {
+        _validateDestinationCallFlag(_bridgeData, _stargateData);
+
+        // get the address of the Stargate pool/router contract
+        address routerAddress = _getRouterAddress(_stargateData.assetId);
+
+        // check if NATIVE or ERC20
+        uint256 msgValue = _stargateData.fee.nativeFee;
+        if (LibAsset.isNativeAsset(_bridgeData.sendingAssetId)) {
+            // NATIVE
+            // add minAmount to msgValue
+            msgValue += _bridgeData.minAmount;
+        } else {
+            // ERC20
+            // check current allowance to router
+            ERC20 sendingAsset = ERC20(_bridgeData.sendingAssetId);
+            uint256 currentAllowance = sendingAsset.allowance(
+                address(this),
+                routerAddress
+            );
+            // check if allowance is sufficient
+            if (currentAllowance < _bridgeData.minAmount) {
+                // check if allowance is 0
+                if (currentAllowance != 0) {
+                    sendingAsset.approve(routerAddress, 0);
+                }
+                // set allowance to uintMax
+                sendingAsset.approve(routerAddress, type(uint256).max);
+            }
+        }
+
+        // execute call to Stargate router
+        IStargate(routerAddress).sendToken{ value: msgValue }(
+            _stargateData.sendParams,
+            _stargateData.fee,
+            _stargateData.refundAddress
+        );
+
+        emit LiFiTransferStarted(_bridgeData);
+    }
+
+    /// PRIVATE HELPER METHODS ///
+    /// @dev makes sure that calldata contains payload for destination when bridging tokens+destCall
+    function _validateDestinationCallFlag(
+        ILiFi.BridgeData memory _bridgeData,
+        StargateData calldata _stargateData
+    ) private pure {
+        if (
+            (_stargateData.sendParams.composeMsg.length > 0) !=
+            _bridgeData.hasDestinationCall
+        ) {
+            revert InformationMismatch();
+        }
+    }
+
+    /// @dev returns the pool-/routerAddress for a given asset(Id), reverts if no address can be found
+    function _getRouterAddress(
+        uint16 assetId
+    ) private returns (address routerAddress) {
+        // get the router-/pool address through the TokenMessaging contract
+        routerAddress = tokenMessaging.stargateImpls(assetId);
+        if (routerAddress == address(0)) revert InvalidAssetId(assetId);
+    }
+
+    receive() external payable {}
```

Comment:
I don't think we need this as the diamond already has it.

Reply:
I added it so that the facet is also usable as standalone contract.
It doesnt hurt, right?

Reply:
I have removed it